compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1318,6 +1318,22 @@ private boolean validateErrorVariable(BLangErrorVariable errorVariable) {
             Map<String, BField> fieldMap = recordType.fields.stream()
                     .collect(Collectors.toMap(f -> f.name.value, f -> f));
 
+            if (isReasonSpecified(errorVariable)
+                    && !errorVariable.reasonVarPrefixAvailable
+                    && errorVariable.reasonMatchConst == null
+                    && errorVariable.isInMatchStmt) {
```

Comment:
What I originally meant is whether `errorVariable.isInMatchStmt` should be the first check in the if block?

Reply:
No preferences from me, will move up